app/src/main/java/io/neurolab/activities/PinLayoutActivity.java

```diff
@@ -33,16 +46,62 @@ protected void onCreate(Bundle savedInstanceState) {
+        colorMap = findViewById(R.id.img_neurolab_color_map);
+        colorMap.setImageDrawable(ResourcesCompat.getDrawable(getResources(),
+                frontSide ? R.drawable.neurolab_front_colormap : R.drawable.neurolab_back_colormap, null));
+        pinImageView.setImageDrawable(ResourcesCompat.getDrawable(getResources(), frontSide ? R.drawable.front_pin_layout : R.drawable.back_pin_layout, null));
 
         pinImageView.setOnTouchListener(this);
+
+        populatePinDetails();
+    }
+
+    private void populatePinDetails() {
+        pinDetails.add(new PinDetails("CLK", "Clock pin to clock data out of data pin", Color.parseColor("#6b40a9"), Color.parseColor("#dc1616")));
+        pinDetails.add(new PinDetails("Rx", "Receiver pin for UART communication", Color.parseColor("#4372a2"), Color.parseColor("#1616dc")));
+        pinDetails.add(new PinDetails("Tx", "Transmitter pin for UART communication", Color.parseColor("#4372a2"), Color.parseColor("#37dc16")));
+        pinDetails.add(new PinDetails("GND", "Ground pin (0 V)", Color.parseColor("#ff4040"), Color.parseColor("#16dcda")));
+        pinDetails.add(new PinDetails("MISO", "SPI interface pin - Master in slave out", Color.parseColor("#ffe040"), Color.parseColor("#b01498")));
+        pinDetails.add(new PinDetails("MOSI", "SPI interface pin - Master out slave in", Color.parseColor("#7d5840"), Color.parseColor("#890b41")));
+        pinDetails.add(new PinDetails("CS", "Chip select pin", Color.parseColor("#7d5840"), Color.parseColor("#ff0b41")));
+        pinDetails.add(new PinDetails("C20", "Feedthrough Cap 0603 20% 50V", Color.parseColor("#7d5840"), Color.parseColor("#ff410b")));
+        pinDetails.add(new PinDetails("C21", "Feedthrough Cap 0603 20% 50V", Color.parseColor("#7d5840"), Color.parseColor("#ff410b")));
+        pinDetails.add(new PinDetails("C24", "Capacitor 16V 10%", Color.parseColor("#ff4040"), Color.parseColor("#d25c3c")));
+        pinDetails.add(new PinDetails("C25", "Capacitor 16V 10%", Color.parseColor("#4372a2"), Color.parseColor("#5c5c3c")));
+        pinDetails.add(new PinDetails("R7", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R17", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R18", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R25", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R34", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R8", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
```

Comment:
it's "Pin descriptions"! @jddeep do you understand this feature? Did you even try the PSLab Android app?

Reply:
These are basically resistance measuring pins...so should I simply put `Resistance measurement pin` as description (similar in PSLab)

Reply:
R8 is a resistor. How do you see it as a **pin**?

Reply:
Just try out the PSLab app and see. This issue should not involve such discussion.